### PR TITLE
Rewrite Debian/Ubuntu packaging

### PR DESCRIPTION
**Pull request by stephankramer**
To avoid having to deal with the setup.py install vagueries when installing python for Debian packaging, let pybuild take care of the building and installing of python. Rewrite debian/rules to make use of debhelper for as much as possible and minimize hackery. Also remove Debian-packaging-specific hacks in Makefile.in

The recipe we are now following is to install everything in debian/tmp (instead of straight into the debian/<package>/ directories), and divide the resulting files up over the packages using debian/<package.install> files. Try to clean up some Lintin errors and warnings in the process.

Note that we are still using distutils in the setup.py files and calling "python setup.py install" which are both deprecated. Only this combination appears to not be hit by https://github.com/pypa/pip/issues/10978 so just leaving as is.

Also introduces:
* testing on both Focal and Jammy
* fix intermittent test failure in build process

### CODE REVIEW

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -22,7 +38,7 @@ jobs:
         run: | 
           echo "Europe/London" > /etc/timezone
           apt-get -y update
-          DEBIAN_FRONTEND=noninteractive apt-get -y install gfortran g++ python3-setuptools python3-all-dev debhelper dh-python texlive python3-junit.xml
+          DEBIAN_FRONTEND=noninteractive apt-get -y install gfortran g++ python3-setuptools python3-pip python3-all-dev debhelper dh-python texlive python3-junit.xml
```
Comment (cianwilson) [inline .github/workflows/ubuntu.yml:41]: Not a problem but curious if the pip installation is still necessary or leftover from when you were trying it that way?

File: debian/control [hunk 2]
```diff
@@ -2,10 +2,10 @@ Source: spud
 Maintainer: The ICOM team <patrick.farrell@imperial.ac.uk>
 Uploaders: David Ham <david.ham@imperial.ac.uk>
 Section: science
-Priority: extra
-Build-Depends: gfortran (>=4.2), g++ (>=4.2), python3-setuptools (>= 0.6b3-1), python3-all-dev (>= 2.3.5-11), debhelper (>= 5.0.38), dh-python, texlive
+Priority: optional
+Build-Depends: gfortran (>=4.2), g++ (>=4.2), python3-setuptools (>= 39), python3-pip, python3-all-dev , debhelper (>= 5.0.38), dh-python, texlive
```
Comment (cianwilson) [inline debian/control:6]: Same question about pip, is it a required dependency?

**Review by cianwilson — APPROVED** [review]
Looks great.  This is a huge improvement and will help me immensely on a few machines where I've run into problems with this already.  Thanks @stephankramer !

Couple of quick questions in the comments but happy to approve regardless.

File: .github/workflows/ubuntu.yml (hunk 1 above)
Comment (stephankramer) [reply .github/workflows/ubuntu.yml]: Removed now

File: debian/control (hunk 2 above)
Comment (stephankramer) [reply debian/control]: No indeed. Removed

**Comment by stephankramer** [comment]
Thanks, good point.
In the mean time I've also added testing of the package building to the actions CI

**Review by cianwilson — APPROVED** [review]
Looks good!

**Pull request merged**